### ISSUE TITLE
Bad type in conditional expression - java.util.UUID cannot be converted to java.util.List<java.util.UUID>

### ISSUE THREAD

**Issue by gusilva** [graphql-java-generator/graphql-maven-plugin-project#51]
In my graphql schema I have a mutation which receives a list of Ids as args. 

```
type Mutation {
    deleteSnacks(id: [ID]) : Boolean
}
```

When I run: `mvn clean install`, it returns the following error:
```
[INFO] -------------------------------------------------------------<br />
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/gustavo/Documents/demo.graphql-plugin-mvn/target/generated-sources/graphql-maven-plugin/com/generated/graphql/GraphQLDataFetchers.java:[132,117] incompatible types: bad type in conditional expression
    java.util.UUID cannot be converted to java.util.List<java.util.UUID>
[INFO] 1 error
[INFO] -------------------------------------------------------------
```

The following method is generated in GraphQLDataFetchers class:

```
public DataFetcher<Boolean> dataFetchersDelegateMutationDeleteSnacks() {
		return dataFetchingEnvironment -> {
			List<UUID> id = (dataFetchingEnvironment.getArgument("id") == null) ? null : UUID.fromString(dataFetchingEnvironment.getArgument("id"));
  
			Boolean ret = null;
			try {
				ret = dataFetchersDelegateMutation.deleteSnacks(dataFetchingEnvironment, id);
			} catch (NoSuchElementException e) {
				// There was no items in the Optional
			}

			if (ret != null)
				logger.debug("deleteSnacks: 1 result found");
			else
				logger.debug("deleteSnacks: no result found");

			return ret;
		};
	}
```

As you can see, the generated method is not fetching the list of UUIDs. 

The repo below, reproduces the error:
https://github.com/gusilva/graphql-maven-demo

**Comment by etienne-sf** [comment]
Hum, hum,


I'm precisely working on list serialization and deserialization.

I'lll add this in my test cases. I'm almost done, so a new release with this issue corrected should be corrected this week end or early next week.

Etienne

**Comment by gusilva** [comment]
Awesome!! thank you @etienne-sf !

> Hum, hum,
> 
> I'm precisely working on list serialization and deserialization.
> 
> I'lll add this in my test cases. I'm almost done, so a new release with this issue corrected should be corrected this week end or early next week.
> 
> Etienne

Awesome!! thank you @etienne-sf !

**Comment by gusilva** [comment]
my bad! By mistake I closed it

**Comment by etienne-sf** [comment]
Yes, you can keep it open
:) 

I confirm the issue.
And I confirm that my recent work on the serialization/deserialization has solved it. You can test it on the master branch, if you wish.
I'll add a specific test case, as this use case is a standard one: it must work.

Etienne

**Comment by etienne-sf** [comment]
I released yesterday the 1.11 version.


Can you confirm that your issue is solved, from your point of view ?

Etienne

**Comment by gusilva** [comment]
> I released yesterday the 1.11 version.
> 
> Can you confirm that your issue is solved, from your point of view ?
> 
> Etienne

The issue is solved. I've just confirmed with the version 1.11! Again, tks @etienne-sf
